src/claimer.py

```diff
@@ -59,6 +59,12 @@ def claim_product(api_client, recaptcha_solution):
             'dateTo': (utc_today + dt.timedelta(days=1)).isoformat()
         }
     )
+    """Traitement de l'absence d'offre"""
+    offer_count=offer_response.json().get('count')
+    if offer_count == 0:
+        logger.info(" Pas d'offre PACKT freelearning disponible ce jour ")
+        raise Exception(" Pas d'offre PACKT freelearning disponible ce jour ")
```

Comment:
I believe we shouldn't raise general exception, I'd rather create a new custom exception or return an empty value (we need to distinguish it from the case when a Packt produck has already been claimed) and handle it further in `packtPublishingFreeEbook.py`.

I'm not sure which way is better though or easier to handle, but I think that custom exception will be easier to handle.

Reply:
You get a message "Today's free Packt ebook grabbing has failed with exception:  : There is no Free Learning offer right now". Not ideal, but meaningfull while reading.